bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/adapters/read/TestResultRowAdapter.java

```diff
@@ -0,0 +1,160 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.hbase.adapters.read;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.bigtable.data.v2.models.RowAdapter;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ByteString;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class TestResultRowAdapter {
+
+  private static final ByteString ROW_KEY_ONE = ByteString.copyFromUtf8("one");
+  private static final String COL_FAMILY = "cf";
+  private static final ByteString QUAL_ONE = ByteString.copyFromUtf8("q1");
+  private static final ByteString QUAL_TWO = ByteString.copyFromUtf8("q2");
+  private static final List<String> LABELS = ImmutableList.of("label-a", "label-b");
+  private static final ByteString VALUE = ByteString.copyFromUtf8("value");
+
+  private ResultRowAdapter underTest = new ResultRowAdapter();
+
+  @Test
+  public void testWithSingleCellRow() {
+    RowAdapter.RowBuilder<Result> rowBuilder = underTest.createRowBuilder();
+
+    rowBuilder.startRow(ROW_KEY_ONE);
+    rowBuilder.startCell(COL_FAMILY, QUAL_ONE, 10000L, Collections.<String>emptyList(), 0);
+    rowBuilder.cellValue(VALUE);
+    rowBuilder.finishCell();
+    rowBuilder.startCell(COL_FAMILY, QUAL_TWO, 20000L, LABELS, 0);
+    rowBuilder.cellValue(VALUE);
+    rowBuilder.finishCell();
+
+    Result expected =
+        Result.create(
+            ImmutableList.<Cell>of(
+                new RowCell(
+                    ROW_KEY_ONE.toByteArray(),
+                    COL_FAMILY.getBytes(),
+                    QUAL_ONE.toByteArray(),
+                    10L,
+                    VALUE.toByteArray()),
+                new RowCell(
+                    ROW_KEY_ONE.toByteArray(),
+                    COL_FAMILY.getBytes(),
+                    QUAL_TWO.toByteArray(),
+                    20L,
+                    VALUE.toByteArray(),
+                    LABELS)));
+    assertResult(expected, rowBuilder.finishRow());
+    assertEquals(ROW_KEY_ONE, underTest.getKey(expected));
+  }
+
+  @Test
+  public void testWhenSplitCell() {
+    ByteString valuePart1 = ByteString.copyFromUtf8("part-1");
+    ByteString valuePart2 = ByteString.copyFromUtf8("part-2");
+    ByteString valuePart3 = ByteString.copyFromUtf8("part-3");
+
+    RowAdapter.RowBuilder<Result> rowBuilder = underTest.createRowBuilder();
+    rowBuilder.startRow(ROW_KEY_ONE);
+    rowBuilder.startCell(COL_FAMILY, QUAL_ONE, 10000L, Collections.<String>emptyList(), 0);
```

Comment:
This looks wrong, why is size 0 for a split cell?

Reply:
This was my overlook because I had not checked against the `size` parameter in this draft.